extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/blocking/inheritance/BlockingFromTransactionalTest.java

```diff
@@ -0,0 +1,106 @@
+package io.quarkus.grpc.server.blocking.inheritance;
+
+import io.quarkus.grpc.GrpcClient;
+import io.quarkus.grpc.GrpcService;
+import io.quarkus.grpc.test.blocking.inheritance.BlockingTestService;
+import io.quarkus.grpc.test.blocking.inheritance.BlockingTestServiceGrpc;
+import io.quarkus.grpc.test.blocking.inheritance.InheritenceTest;
+import io.quarkus.test.QuarkusUnitTest;
+import io.smallrye.common.annotation.Blocking;
+import io.smallrye.common.annotation.NonBlocking;
+import io.smallrye.mutiny.Uni;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import javax.transaction.Transactional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BlockingFromTransactionalTest {
+    private static final String BLOCKING = "blocking";
+    private static final String NON_BLOCKING = "nonblocking";
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(
+                    () -> ShrinkWrap.create(JavaArchive.class)
+                            .addPackage(BlockingTestServiceGrpc.class.getPackage())
+                            .addClasses(BlockingTestService.class));
+
+    @GrpcClient
+    BlockingTestService client;
+
+    @Test
+    @Timeout(5)
+    void shouldBlockOnInheritedClassLevelTransactional() {
+        assertThat(client.notOverriden1(message()).await().indefinitely().getText()).isEqualTo(BLOCKING);
+    }
+
+    @Test
+    @Timeout(5)
+    void shouldNotBlockOnInheritedTransactionalMarkedNonBlocking() {
+        assertThat(client.notOverriden2(message()).await().indefinitely().getText()).isEqualTo(NON_BLOCKING);
+    }
+
+    @Test
+    @Timeout(5)
+    void shouldNotBlockOnOverridenTransactionalMarkedNonBlocking() {
+        assertThat(client.overriden1(message()).await().indefinitely().getText()).isEqualTo(NON_BLOCKING);
+    }
+
+    @Test
+    @Timeout(5)
+    void shouldBlockOnOverridenTransactional() {
```

Comment:
this test fails, so it seems the sixth point of the description does not work, i.e. `blocking, if the service class or any of its base classes has a Transaction annotation.`
Or am I missing something?